src/engine/input/InputManager.cs

```diff
@@ -57,7 +57,7 @@ public static void RegisterReceiver(object instance)
         // TODO: check if there is some alternative faster approach to registering instances
         foreach (var inputAttribute in staticInstance
             .attributes
-            .Where(p => p.Key.Method.DeclaringType == instance.GetType()))
+            .Where(p => p.Key.Method.DeclaringType?.IsInstanceOfType(instance) == true))
```

Comment:
What's this change about?

Reply:
It was a bug in the input system.
I define a `RunOnKeyDown` Method in `CheatMenu` (the parent class), therefore `DeclaringType` is `typeof(CheatMenu)`
The actual runtime instance that calls the `RegisterReceiver` is `MicrobeCheatMenu` (the child class), therefore `instance` is a `MicrobeCheatMenu`

I cannot make a direct comparison, because `CheatMenu` is not `MicrobeCheatMenu`. I have to do an `is` operation, which is `IsInstanceOfType` when working with generics.

Reply:
That explains it. Elsewhere I've used the "is assignable from" method on the type. Should this change be also added elsewhere in the input handling?

Reply:
no, I haven't found anything.
Unregister should be fine, because it only compares runtime instances

Reply:
Why was this not changed to use IsAssignableFrom as is used elsewhere in the codebase to determine if something is a compatible type?
```
$ ag IsAssign
src/saving/serializers/ThriveJsonConverter.cs
328:        return typeof(Node).IsAssignableFrom(type) && BaseNodeConverter.IsIgnoredGodotNodeMember(name);

src/saving/serializers/NodeConverter.cs
15:        return typeof(T).IsAssignableFrom(objectType);

src/saving/serializers/BaseNodeConverter.cs
60:        return typeof(Node).IsAssignableFrom(objectType);

src/saving/serializers/CallbackConverter.cs
135:        if (typeof(MulticastDelegate).IsAssignableFrom(objectType))

```

Reply:
The only difference between IsAssignableFrom and IsInstanceOfType is that IsInstanceOfType takes an object and IsAssignableFrom takes a type. Logic wise they are the same, but IsInstanceOfType allows me to omit the .GetType call

Reply:
Okay then it's fine, I didn't consider that you didn't already have the type object to check with.